### ISSUE TITLE
Error in bin/doom

### ISSUE THREAD

**Issue by wantyapps** [doomemacs/doomemacs#6480]
### What did you expect to happen?

I just formatted my computer and moved from Fedora 36 (running doom fine) to ArcoLinux. When I copied and pasted the commands to install, I was expecting it to install the latest Doom Emacs.

### What actually happened?

When running `bin/doom` it crashes with the following error:
```
Debugger entered--Lisp error: (void-variable doom-cli--dump)
  (and doom-cli--dump (equal args '("-")))
  (if (and doom-cli--dump (equal args '("-"))) (doom-cli--dump (progn (doom-cli-load-all) (hash-table-values doom-cli--table))) (let ((context (doom-cli-context-create :prefix prefix))) (let ((doom-cli--context context)) (let ((write-logs-fn (doom-partial #'doom-cli--output-write-logs-h context))) (let ((show-benchmark-fn (doom-partial ... context))) (let ((debugger ...)) (let (...) (let ... ...))))))))
  (let ((args (flatten-list args))) (if (and doom-cli--dump (equal args '("-"))) (doom-cli--dump (progn (doom-cli-load-all) (hash-table-values doom-cli--table))) (let ((context (doom-cli-context-create :prefix prefix))) (let ((doom-cli--context context)) (let ((write-logs-fn (doom-partial ... context))) (let ((show-benchmark-fn ...)) (let (...) (let ... ...))))))))
  run!("doom" nil)
  eval-buffer(#<buffer  *load*> nil "/home/wanty/.emacs.d/bin/doom" nil t)  ; Reading at buffer position 12247
  load-with-code-conversion("/home/wanty/.emacs.d/bin/doom" "/home/wanty/.emacs.d/bin/doom" nil t)
  load("/home/wanty/.emacs.d/bin/doom" nil t)
  command-line-1(("--load" ".emacs.d/bin/doom" "--"))
  command-line()
  normal-top-level()
```

### Describe your attempts to resolve the issue

The error code returned by the program crashing was `255`. Naturally, I went and opened the doom executable to see the documentation about it. It said that its an Uncaught Internal Error

### Steps to reproduce

1. Use the commands provided by the main README to install the latest Doom Emacs. Those commands are:
```
git clone --depth 1 https://github.com/doomemacs/doomemacs ~/.emacs.d
~/.emacs.d/bin/doom install
```
2. See the error

### System Information

(Doom cannot provide the info, as the command isn't working.)

**Comment by hlissner** [comment]
This regression snuck in on d231755bdfa288aababe7b738a5677fe4cb0c6e9 and was fixed in 051144533. You'll need to update Doom manually to overcome it:

```sh
cd ~/.emacs.d
git pull --rebase
doom sync -u
```

**Comment by wantyapps** [comment]
Thank you! Before posting the issue, I tried pulling with many commands (including this one) and it told me my clone was the latest. I guess you guys fixed it overnight hehe
